ui-common/src/components/UI/SelectList.tsx

```diff
@@ -1,4 +1,4 @@
-import React, { useMemo } from "react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
```

Comment:
change headerChecked state only from a single place - create useEffect and check diff between checkedValues and filteredValues - only if equal, setHeaderChecked(true), else, setHeaderChecked(false)